composer/core/callback.py

```diff
@@ -58,246 +46,29 @@ def run_event(self, event: Event, state: State, logger: Logger) -> None:
-        """Called on the :attr:`~Event.BATCH_END` event.
-
-        Args:
-            state (State): The global state.
-            logger (Logger): The logger.
-        
-        """
-        del state, logger  # unused
-        pass
-
-    def epoch_end(self, state: State, logger: Logger) -> None:
-        """Called on the :attr:`~Event.EPOCH_END` event.
-
-        Args:
-            state (State): The global state.
-            logger (Logger): The logger.
-        
-        """
-        del state, logger  # unused
-        pass
-
-    def training_end(self, state: State, logger: Logger) -> None:
-        """Called on the :attr:`~Event.TRAINING_END` event.
-
-        Args:
-            state (State): The global state.
-            logger (Logger): The logger.
-        
-        """
-        del state, logger  # unused
-        pass
-
-    def eval_start(self, state: State, logger: Logger) -> None:
-        """Called on the :attr:`~Event.EVAL_START` event.
-
-        Args:
-            state (State): The global state.
-            logger (Logger): The logger.
-        
-        """
-        del state, logger  # unused
-        pass
-
-    def eval_batch_start(self, state: State, logger: Logger) -> None:
-        """Called on the :attr:`~Event.EVAL_BATCH_START` event.
-
-        Args:
-            state (State): The global state.
-            logger (Logger): The logger.
-        
-        """
-        del state, logger  # unused
-        pass
-
-    def eval_before_forward(self, state: State, logger: Logger) -> None:
-        """Called on the :attr:`~Event.EVAL_BATCH_FORWARD` event.
-
-        Args:
-            state (State): The global state.
-            logger (Logger): The logger.
-        
-        """
-        del state, logger  # unused
-        pass
-
-    def eval_after_forward(self, state: State, logger: Logger) -> None:
-        """Called on the :attr:`~Event.EVAL_AFTER_FORWARD` event.
-
-        Args:
-            state (State): The global state.
-            logger (Logger): The logger.
-        
-        """
-        del state, logger  # unused
-        pass
-
-    def eval_batch_end(self, state: State, logger: Logger) -> None:
-        """Called on the :attr:`~Event.EVAL_BATCH_END` event.
-
-        Args:
-            state (State): The global state.
-            logger (Logger): The logger.
-        
-        """
-        del state, logger  # unused
-        pass
-
-    def eval_end(self, state: State, logger: Logger) -> None:
-        """Called on the :attr:`~Event.EVAL_END` event.
-
-        Args:
-            state (State): The global state.
-            logger (Logger): The logger.
-        
-        """
-        del state, logger  # unused
-        pass
-
 
 class RankZeroCallback(Callback, abc.ABC):
     """Base class for callbacks that only run on the rank zero process.
+
+    Subclasses should override :meth:`_run_event`
+    (**not** `run_event`) to run in response
+    to given :class:`Event` invocations.
```

Comment:
I like this concept, but see no reason to limit it to `RankZeroCallback`. Could it be imposed for `Callback` instead? It would mean that calling `super().run_event` is no longer necessary.

Reply:
Super calls no longer necessary.